front/src/applications/operationalStudies/hooks/useTrackSections.ts

```diff
@@ -0,0 +1,47 @@
+import { useState, useCallback } from 'react';
+
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import type { InfraObjectWithGeometry } from 'common/api/osrdEditoastApi';
+import type { AppDispatch } from 'store';
+
+export default function useCachedTrackSections(infraId: number, dispatch: AppDispatch) {
+  const [trackIds, setTrackIds] = useState<Set<string>>(new Set());
+  const [trackSections, setTrackSections] = useState<InfraObjectWithGeometry[]>([]);
+  const [loading, setLoading] = useState<boolean>(false);
+
+  const loadTracksSectionsByIds = useCallback(
+    async (newTrackIds: string[]) => {
+      const uniqueNewIds = newTrackIds.filter((id) => !trackIds.has(id));
+      if (uniqueNewIds.length === 0) return;
+
+      const fetchedSections = await dispatch(
+        osrdEditoastApi.endpoints.postInfraByInfraIdObjectsAndObjectType.initiate({
+          infraId,
+          objectType: 'TrackSection',
+          body: uniqueNewIds,
+        })
+      ).unwrap();
```

Comment:
Perhaps usequery since we only fetch? I'm not sure I fully understand the implications of the choice of using dispatch, usemutation or usequery though, so I'll defer to your judgement 

(Nitpick : that loading will encompass less code though, but the sync code should be instantaneous so it probably doesn't matter at all.)

Reply:
Like some other endpoints from the api, we fetch here but in reality, as we need to pass a body to the call, it's technically a post so we can only do a `useMutation` :)

If we want to use it with a query, we need to override the endpoint like we did for other of them in this file : `openapi-editoast-config.ts`

Reply:
Thanks for the explanation! I ended up going with useMutation.